### PR TITLE
Fluent 2 Theme for V8

### PR DESCRIPTION
**Pull request by mltejera**
Adds Fluent2 Theme for V8 functions, styles, colors and all that goes with the theme.

Adds demo page showing all the things that have changed.

This is just the MVP for what we've promised our partners. In a follow up PR, I'm going to add functionality to change the brand ramp.

image
image

### CODE REVIEW

File: packages/react-migration-v8-v9/src/stories/ThemeShim/createFluent2ThemeForV8/DemoComponents/ThemeDemoPage.tsx [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+    setSelectedTheme(Fluent2ForV8LightTheme);
+  };
+
+  const handleDarkSelection = () => {
+    setSelectedTheme(Fluent2ForV8DarkTheme);
+  };
+
+  return (
+    <ThemeProvider theme={selectedTheme}>
+      <div className={rootStyles}>
```
Comment (mltejera) [inline packages/react-migration-v8-v9/src/stories/ThemeShim/createFluent2ThemeForV8/DemoComponents/ThemeDemoPage.tsx:55]: Would it be cool if I added a button to submit bugs here?

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/5079ab52943dded035d7bf7348d1c2762271ae9d'>5079ab52943dded035d7bf7348d1c2762271ae9d</a></sub>
<div id="bundleSizeComment9423"/>

File: packages/react-migration-v8-v9/src/components/Theme/Fluent2ThemeForV8/fluent2ForV8.types.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import { IEffects, ITheme } from '@fluentui/react';
+
+// this exists so that both theme name and theme specifications can be accessed using fixed keys for easy iteration
+export interface INamedTheme {
+  humanFriendlyName: string;
+  themeValues: ITheme;
+}
+
+export interface IExtendedEffects extends IEffects {
+  roundedCorner8?: string;
```
Comment (mltejera) [inline packages/react-migration-v8-v9/src/components/Theme/Fluent2ThemeForV8/fluent2ForV8.types.ts:10]: double check visually that the 8 is getting passed through

File: packages/react-migration-v8-v9/src/stories/ThemeShim/createFluent2ThemeForV8/DemoComponents/FluentV8ComponentSamples.tsx [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+    },
+    iconOnly: true,
+    onClick: () => console.log('List'),
+  },
+];
+
+export const FluentComponentSamples: FunctionComponent = () => {
+  const [isContentDisabled, setIsContentDisabled] = React.useState<boolean>(false);
+
+  initializeIcons();
```
Comment (mltejera) [inline packages/react-migration-v8-v9/src/stories/ThemeShim/createFluent2ThemeForV8/DemoComponents/FluentV8ComponentSamples.tsx:118]: memoize this so it's not getting called on every render.

File: packages/react-migration-v8-v9/src/components/Theme/Fluent2ThemeForV8/fluent2-for-v8-dark-theme.ts [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+  black: '#ffffff', // Note white and black are inverted in this theme
+  white: '#141414',
+};
+
+const p = fluent2ForV8DarkPalette;
+
+// shortcut since the gray scale isn't part of the v8 palette.
+const grey36 = '#5C5C5C';
+
+const extendedSemanticColors: Partial<ISemanticColors> = {
```
Comment (mltejera) [inline packages/react-migration-v8-v9/src/components/Theme/Fluent2ThemeForV8/fluent2-for-v8-dark-theme.ts:50]: pick a more specific name for this

File: packages/react-migration-v8-v9/src/components/Theme/Fluent2ThemeForV8/fluent2-for-v8-light-theme.ts [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+  elevation16: '0px 8px 16px rgba(0, 0, 0, 0.14), 0px 0px 2px rgba(0, 0, 0, 0.12)',
+  elevation64: '0px 32px 64px rgba(0, 0, 0, 0.24), 0px 0px 8px rgba(0, 0, 0, 0.2)',
+  roundedCorner2: '2px',
+  roundedCorner4: '4px',
+  roundedCorner6: '6px',
+  roundedCorner8: '8px',
+  roundedCornerCircle: '50%',
+};
+
+// Warnings and errors
```
Comment (mltejera) [inline packages/react-migration-v8-v9/src/components/Theme/Fluent2ThemeForV8/fluent2-for-v8-light-theme.ts:20]: remove

File: packages/react-migration-v8-v9/src/components/Theme/Fluent2ThemeForV8/fluent2ForV8.types.ts [hunk 6]
```diff
@@ -0,0 +1,12 @@
+import { IEffects, ITheme } from '@fluentui/react';
+
+// this exists so that both theme name and theme specifications can be accessed using fixed keys for easy iteration
+export interface INamedTheme {
```
Comment (mltejera) [inline packages/react-migration-v8-v9/src/components/Theme/Fluent2ThemeForV8/fluent2ForV8.types.ts:4]: I doon't think this interface is necesary anymore

File: packages/react-migration-v8-v9/src/components/Theme/Fluent2ThemeForV8/fluent2-for-v8-dark-theme.ts [hunk 7]
```diff
@@ -0,0 +1,88 @@
+/* eslint-disable @typescript-eslint/no-namespace */
```
[the remark this answers is not in the document]
Comment (mltejera) [reply packages/react-migration-v8-v9/src/components/Theme/Fluent2ThemeForV8/fluent2-for-v8-dark-theme.ts:1]: whoops - missed this one, sorry! Wilco :-)

File: packages/react-migration-v8-v9/src/components/Theme/Fluent2ThemeForV8/fluent2ForV8.types.ts (hunk 2 above)
Comment (mltejera) [reply packages/react-migration-v8-v9/src/components/Theme/Fluent2ThemeForV8/fluent2ForV8.types.ts:5]: yep - sure is! :-)

**Review by GeoffCoxMSFT — APPROVED** [review]

File: packages/react-migration-v8-v9/src/stories/ThemeShim/createFluent2ThemeForV8/DemoComponents/ThemeDemoPage.tsx (hunk 1 above)
Comment (JustSlone) [reply packages/react-migration-v8-v9/src/stories/ThemeShim/createFluent2ThemeForV8/DemoComponents/ThemeDemoPage.tsx:56]: Yeah, that's a good idea, for reference this is what it looks like on the main doc site, just a primary button that is a link.

image

**Comment by JustSlone** [comment]
Awesome work 🤩

@mltejera @GeoffCoxMSFT are there particular parts of this epic PR that you'd like 👀 / review on?

**Pull request closed without merging**